awx/main/tests/unit/test_tasks.py

```diff
@@ -318,15 +318,22 @@ def test_bwrap_virtualenvs_are_readonly(self):
 
     def test_created_by_extra_vars(self):
         self.instance.created_by = User(pk=123, username='angry-spud')
-        self.task.run(self.pk)
 
-        assert self.run_pexpect.call_count == 1
-        call_args, _ = self.run_pexpect.call_args_list[0]
-        args, cwd, env, stdout = call_args
-        assert '"tower_user_id": 123,' in ' '.join(args)
-        assert '"tower_user_name": "angry-spud"' in ' '.join(args)
-        assert '"awx_user_id": 123,' in ' '.join(args)
-        assert '"awx_user_name": "angry-spud"' in ' '.join(args)
+        def run_pexpect_side_effect(*args, **kwargs):
+            args, cwd, env, stdout = args
+            extra_vars = {}
+            for chunk in args:
+                if chunk.startswith('@/tmp/'):
+                    with open(chunk.strip('@'), 'r') as f:
+                        extra_vars.update(json.load(f))
+            assert extra_vars['tower_user_id'] == 123
+            assert extra_vars['tower_user_name'] == "angry-spud"
+            assert extra_vars['awx_user_id'] == 123
+            assert extra_vars['awx_user_name'] == "angry-spud"
+            return ['successful', 0]
+
+        self.run_pexpect.side_effect = run_pexpect_side_effect
```

Comment:
@wenottingham your diff had `self.run_pexpect_side_effect` rather than `self.run_pexpect.side_effect`; that slight typo was causing these tests to _not_ run the side effect function (which includes all the test assertions), thus causing the tests to silently pass, even though it wasn't working.